runtime/sys.c

```diff
@@ -442,9 +442,13 @@ CAMLprim value caml_sys_system_command(value command)
     caml_sys_error(command);
   }
   buf = caml_stat_strdup_to_os(String_val(command));
-  caml_enter_blocking_section ();
-  status = system_os(buf);
-  caml_leave_blocking_section ();
+  #if HAS_SYSTEM
+    caml_enter_blocking_section ();
+    status = system_os(buf);
+    caml_leave_blocking_section ();
+  #else
+    caml_invalid_argument("Sys.command not implemented");
+  #endif /* HAS_SYSTEM */
```

Comment:
Shouldn't the `HAS_SYSTEM` define protect the full function definition? ie
```
#ifdef HAS_SYSTEM
...
#else
CAMLprim value caml_sys_system_command(value command)
{
  caml_invalid_argument("Sys.command not implemented");
}
#endif
```

Reply:
I don't think so, because then the symbol will be missing, also I was following the pattern on the same file at `getcwd`

https://github.com/ocaml/ocaml/blob/83a06c3d1582cc582ae575a1145edca1432476d9/runtime/sys.c#L327

Reply:
Which symbol? `caml_sys_system_command` will always be defined (either with its good definition or with the dummy definition that raises `Invalid_argument`).

Note that the current way of doing this, you are leaking `buf` in each function call.

Reply:
Oh that makes sense, so maybe we should change `getcwd` too?

But it's not leaking `caml_stat_free(buf);` is called in the next line. at 452.

Reply:
452 is not executed if you call `caml_invalid_argument` :)

Reply:
In `getcwd` there is no memory leak so it can be left as it is I think.

Reply:
Fixed, I had forgot that caml_invalid_argument raises an exception, addressed